### DISCUSSION TITLE
Add a possibility to update dependencies to second most latest version

### DISCUSSION THREAD

**Discussion by furkatgofurov7** [renovatebot/renovate#22214] [Suggest an Idea]
**What would you like Renovate to be able to do?**
Hi, not sure if this is already possible, at least was not able to find it out for now. Basically, we would like Renovate to be able to update version of dependency one lower version from the latest version. i.e. as is now, latest k8s version is v1.21.3 and renovate should update the k8s_version to v1.21.2 instead of the latest. 

We have a case, where in CI we run couple of test cases in ansible, and to be specific a test case for upgrade, where we try to upgrade k8s version from one version (usually the version prior to the latest) to the latest version. So, we defined different variables for that, i.e `from_k8s_version=v1.21.1` and `to_k8s_version=v1.21.2`. And we would like to add them to renovate json so that, whenever, there is a new version of kubernetes (i.e  v1.21.3), we could update `from_k8s_version` to v1.21.2 which is one version prior to the latest (but not sure if renovate is smart enough to be told to update this var to N-1) and  `to_k8s_version` to v1.21.3 which works perfectly with the current capabilities of renovate.
**Did you already have any implementation ideas?**
No

**Comment by rarkins** [comment]
It's not currently possible that I can think of, and it's a bit of a hack.. but I see why you're doing it so I'd like to try to support it. I'm thinking that we could reuse our `followTag` feature for this instead of creating a new one and maybe have a special value of "N-1" which means "always set this dependency to be the second most latest version"

**Comment by furkatgofurov7** [comment]
Thanks for reply @rarkins! I am happy to go around this for now even though it is not supported, would you mind giving a bit more of a context how can we achieve supporting this feature with `followTag`? i.e examples how it is used for same purposes or anything that I can get a better idea? Thank you

**Comment by rarkins** [comment]
It would need development/a PR to support. The current logic for followTag is here: https://github.com/renovatebot/renovate/blob/aa7f1cb95293c94d632a333e5973ce6c4c53cc18/lib/workers/repository/process/lookup/index.ts#L100-L115 and any new logic would likely be inserted there. The main challenge is that at that point in the code, allVersions is indeed all versions, so if there were some beta/pre-release versions then that would be selected rather than "latest stable minus 1". To implement "latest stable minus 1" would mean the logic has to come later in the file

**Comment by machine424** [comment]
Assuming renovate can update versions in its own config files.
One way could be to separate the config into two: the one for `second most latest version` would have `allowedVersions: < LATEST_VRSION` https://docs.renovatebot.com/configuration-options/#allowedversions
then using https://docs.renovatebot.com/modules/manager/regex/ you can tell renovate to keep LATEST_VRSION up to date.
Not sure if all of this could be done atomically though/in the same PR...

**Comment by chokoswitch** [comment]
Just wanted to add a somewhat different, but possibly more common, use case - for Go library packages that want to target the current supported versions of Go, they have to have the previous _minor_ version in `go.mod` because Go supports two minor versions at any time, e.g. currently since Go 1.25.0 is the latest, go.mod will usually be Go 1.24.0. While it's not the end of the world to use 1.24.x being the latest in that minor, it's generally considered bad practice for libraries since it can force updates on users too early.

All that to say that this would need the ability to set to the second latest minor version, not second latest version absolutely which would often be a patch on the latest minor. New minor versions are released every half a year, not often so this is definitely not a high priority, though updating many repos at that time can be some work, so definitely fun if it was supported.

**Reply by jamietanna** [reply]
Regarding Go - https://github.com/renovatebot/renovate/issues/42567 will make it possible to tell Renovate you only want Go module updates within your specified `go` directive
